compiler/rustc_middle/src/ty/inhabitedness/mod.rs

```diff
@@ -112,7 +112,11 @@ impl<'tcx> Ty<'tcx> {
                 InhabitedPredicate::True
             }
             Never => InhabitedPredicate::False,
-            Param(_) | Alias(ty::Projection, _) => InhabitedPredicate::GenericType(self),
+            // FIXME(inherent_associated_types): It's unclear if the substs passed to us are of
+            // the correct format (i.e. don't contain parent substs). Double-check this.
+            Param(_) | Alias(ty::Projection | ty::Inherent, _) => {
```

Comment:
Couldn't come up with a snippet that reaches this code path without encountering cycle errors first. Will try again.

Reply:
Can you put a bug here instead then? With a FIXME that if this is ever hit, we should deal with it?